core/base_algo.py

```diff
@@ -371,26 +372,31 @@ def get_remaining_categories(self) -> Set:
         return ALL_CATEGORIES - categories_in_file
 
     def _update_results_csv(self, values_dict) -> None:
-        if self.args.category in set(self.experiment_results_manager.data.category):
-            mask = (self.experiment_results_manager.data.category != self.args.category)
+        key = [self.args.category, self.model_name]
+        if self.experiment_results_manager.data[['category', 'model_name']].isin(key).any(axis=1).any():
+            # Key match found, overwriting
+            mask = (self.experiment_results_manager.data[['category', 'model_name']] != key)
```

Comment:
i'll believe that you tested it.

Reply:
I tested it in ipython, creating custom dataframes